src/openrct2/scripting/bindings/entity/ScStaff.cpp

```diff
@@ -104,23 +105,112 @@ namespace OpenRCT2::Scripting
         }
     }
 
-    uint8_t ScStaff::costume_get() const
+    static const DukEnumMap<PeepSpriteType> availableHandymanCostumes({
+        { "handyman", PeepSpriteType::Handyman },
+    });
+
+    static const DukEnumMap<PeepSpriteType> availableMechanicCostumes({
+        { "mechanic", PeepSpriteType::Mechanic },
+    });
+
+    static const DukEnumMap<PeepSpriteType> availableSecurityCostumes({
+        { "security1", PeepSpriteType::Security },
+        { "security2", PeepSpriteType::SecurityAlt },
+    });
+
+    static const DukEnumMap<PeepSpriteType> availableEntertainerCostumes({
+        { "none", PeepSpriteType::Normal },
+        { "panda", PeepSpriteType::EntertainerPanda },
+        { "tiger", PeepSpriteType::EntertainerTiger },
+        { "elephant", PeepSpriteType::EntertainerElephant },
+        { "roman", PeepSpriteType::EntertainerRoman },
+        { "gorilla", PeepSpriteType::EntertainerGorilla },
+        { "snowman", PeepSpriteType::EntertainerSnowman },
+        { "knight", PeepSpriteType::EntertainerKnight },
+        { "astronaut", PeepSpriteType::EntertainerAstronaut },
+        { "bandit", PeepSpriteType::EntertainerBandit },
+        { "sheriff", PeepSpriteType::EntertainerSheriff },
+        { "pirate", PeepSpriteType::EntertainerPirate },
+    });
+
+    static const DukEnumMap<PeepSpriteType>& costumesByStaffType(StaffType staffType)
     {
+        switch (staffType)
+        {
+            case StaffType::Handyman:
+                return availableHandymanCostumes;
+            case StaffType::Mechanic:
+                return availableMechanicCostumes;
+            case StaffType::Security:
+                return availableSecurityCostumes;
+            case StaffType::Entertainer:
+            default:
+                return availableEntertainerCostumes;
+        }
+    }
+
+    std::vector<std::string> ScStaff::availableCostumes_get() const
+    {
+        std::vector<std::string> availableCostumes{};
         auto peep = GetStaff();
-        if (peep != nullptr && peep->AssignedStaffType == StaffType::Entertainer)
+        if (peep != nullptr)
+        {
+            for (auto& costume : costumesByStaffType(peep->AssignedStaffType))
+            {
+                availableCostumes.push_back(std::string(costume.first));
+            }
+        }
+        return availableCostumes;
+    }
+
+    std::string ScStaff::costume_get() const
+    {
+        auto peep = GetStaff();
+        if (peep == nullptr)
+        {
+            return nullptr;
+        }
+
+        auto& availableCostumes = costumesByStaffType(peep->AssignedStaffType);
+
+        auto costume = availableCostumes.find(peep->SpriteType);
+        if (costume != availableCostumes.end())
         {
-            return peep->GetCostume();
+            return std::string(costume->first);
         }
-        return 0;
+        else
+            return nullptr;
```

Comment:
To do: `throw DukException(...)`?

Reply:
For the getter, I'd say preferable no. For the setter, it may be preferred though instead of silently not doing anything. 🙂

Reply:
Agreed, yes.